packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -553,10 +555,18 @@ export function getEnvForPackageManager({
     env,
   });
 
-  const corepackEnabled = env.ENABLE_EXPERIMENTAL_COREPACK === '1';
-  debug(
-    `Detected ${detectedPackageManager} given lockfileVersion "${lockfileVersion}", package manager cli "${cliType}", and corepack enabled? ${corepackEnabled}: ${newPath}`
-  );
+  const corepackFlagged = env.ENABLE_EXPERIMENTAL_COREPACK === '1';
+  const corepackEnabled = corepackFlagged && packageJsonPackageManager;
+
+  if (corepackEnabled) {
+    debug(
+      `Detected corepack use for "${packageJsonPackageManager}". Not overriding package manager version.`
+    );
+  } else {
+    debug(
+      `Detected ${detectedPackageManager}. Added "${newPath}" to path. Based on assumed package manager "${cliType}", lockfile "${detectedLockfile}", and lockfileVersion "${lockfileVersion}"`
+    );
+  }
```

Comment:
This change only seems to affect the debug logging, won't we need to change this in `getPathOverrideForPackageManager` too?

Reply:
specifically here: https://github.com/vercel/vercel/blob/265f6888a4e9034f0e46feb62b0b8ba367b4a6ed/packages/build-utils/src/fs/run-user-scripts.ts#L683